src/utilcode/clrconfig.cpp

```diff
@@ -297,16 +297,25 @@ BOOL CLRConfig::IsConfigEnabled(const ConfigDWORDInfo & info)
         return FALSE;
 }
 
-
-
 // 
 // Look up a DWORD config value.
 // 
 // Arguments:
-//     * info - see file:../inc/CLRConfig.h for details
-//     
+//     * info - see file:../inc/CLRConfig.h for details.
+//
+//     * useDefaultIfNotSet - if true, fall back to the default value if the value is not set.
+//
+//     * acceptExplicitDefaultFromRegutil - if false, only accept a value returned by REGUTIL if it is
+//           different from the default value. This parameter is useful as a way to preserve existing
+//           behavior.
+//
+//     * result - the result.
+//
+// Return value:
+//     * true for success, false otherwise.
+// 
 // static
-DWORD CLRConfig::GetConfigValue(const ConfigDWORDInfo & info)
+DWORD CLRConfig::GetConfigValue(const ConfigDWORDInfo & info, bool acceptExplicitDefaultFromRegutil, /* [Out] */ bool *isDefault)
 {
```

Comment:
Effectively, there was a bug in the old code where it would reject a value that came from REGUTIL if it matched the default value, even if it was explicitly set. The comment near that code explained that they left it that way for legacy compat reasons (in case someone was depending on that behavior). Since there are too many users of this function to investigate on a case-by-case basis, this switch allows us to maintain the legacy behavior for all the existing callers, and for new areas that we're touching we can use the fixed behavior.

It also means that the new configuration stuff really does what it says (uses an explicit value if it's set, even if it matches a default). You can imagine a scenario where "System.Foo.Whatever" is set to something other than the default. If someone wants to override that and set it back to the default with a REGUTIL setting, they couldn't do that without 'acceptExplicitDefaultFromRegutil'.